### PR TITLE
spytrap-adb: init at 0.3.3

### PR DESCRIPTION
**Pull request by kpcyrd**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sp/spytrap-adb/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  android-tools,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "spytrap-adb";
+  version = "0.3.3";
+
+  src = fetchFromGitHub {
+    owner = "spytrap-org";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/sp/spytrap-adb/package.nix:14]:
```suggestion
    repo = "spytrap-adb";
```

File: pkgs/by-name/sp/spytrap-adb/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    rev = "refs/tags/v${version}";
```
Comment (Aleksanaa) [inline pkgs/by-name/sp/spytrap-adb/package.nix:15]:
```suggestion
    tag = "v${version}";
```

File: pkgs/by-name/sp/spytrap-adb/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    hash = "sha256-CL+MxSzHpOq2MXmsaa9sipQZ06Kkzy4r1eFjUrPSj1E=";
+  };
+
+  cargoHash = "sha256-Utg3ds0nGAUjBfuEjGfkohkmssKIZT1NiH9hiFnW4ps=";
+
+  SPYTRAP_ADB_BINARY = "${android-tools}/bin/adb";
```
Comment (Aleksanaa) [inline pkgs/by-name/sp/spytrap-adb/package.nix:21]:
```suggestion
  env.SPYTRAP_ADB_BINARY = "${android-tools}/bin/adb";
```

File: pkgs/by-name/sp/spytrap-adb/package.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  meta = with lib; {
+    description = "Test a phone for stalkerware using adb and usb debugging to scan for suspicious apps and configuration";
+    homepage = "https://github.com/spytrap-org/spytrap-adb";
+    changelog = "https://github.com/spytrap-org/spytrap-adb/releases/tag/v${version}";
+    license = with licenses; [ gpl3Plus ];
+    maintainers = with maintainers; [ kpcyrd ];
+    platforms = platforms.all;
```
Comment (Aleksanaa) [inline pkgs/by-name/sp/spytrap-adb/package.nix:29]:
```suggestion
  meta = {
    description = "Test a phone for stalkerware using adb and usb debugging to scan for suspicious apps and configuration";
    homepage = "https://github.com/spytrap-org/spytrap-adb";
    changelog = "https://github.com/spytrap-org/spytrap-adb/releases/tag/v${version}";
    license = with lib.licenses; [ gpl3Plus ];
    maintainers = with lib.maintainers; [ kpcyrd ];
    platforms = lib.platforms.all;
```

File: maintainers/maintainer-list.nix
Comment (Aleksanaa) [inline maintainers/maintainer-list.nix:1]: Please separate this to commit `maintainers: add kpcyrd`

**Review by Aleksanaa — COMMENTED** [review]
Only stylistic suggestions

**Comment by kpcyrd** [comment]
@Aleksanaa thanks, I've pushed fixed commits :)

File: pkgs/by-name/sp/spytrap-adb/package.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  env.SPYTRAP_ADB_BINARY = "${android-tools}/bin/adb";
+
+  meta = {
+    description = "Test a phone for stalkerware using adb and usb debugging to scan for suspicious apps and configuration";
+    homepage = "https://github.com/spytrap-org/spytrap-adb";
+    changelog = "https://github.com/spytrap-org/spytrap-adb/releases/tag/v${version}";
+    license = with lib.licenses; [ gpl3Plus ];
+    maintainers = with lib.maintainers; [ kpcyrd ];
+    platforms = lib.platforms.all;
```
Comment (Aleksanaa) [inline pkgs/by-name/sp/spytrap-adb/package.nix:29]: missing `mainProgram`?

Comment (kpcyrd) [reply pkgs/by-name/sp/spytrap-adb/package.nix:30]: thanks, done ✅

**Comment by kpcyrd** [comment]
Friendly ping, is there anything I can do here? :)

**Comment by Aleksanaa** [comment]
@ofborg build spytrap-adb

**Comment by kpcyrd** [comment]
> Started 3d 10h 46m 11s ago

Sorry to bother again, is there any chance the build is stuck in some way?

File: pkgs/by-name/sp/spytrap-adb/package.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchFromGitHub {
+    owner = "spytrap-org";
+    repo = "spytrap-adb";
+    tag = "v${version}";
+    hash = "sha256-CL+MxSzHpOq2MXmsaa9sipQZ06Kkzy4r1eFjUrPSj1E=";
+  };
+
+  cargoHash = "sha256-Utg3ds0nGAUjBfuEjGfkohkmssKIZT1NiH9hiFnW4ps=";
+
+  env.SPYTRAP_ADB_BINARY = "${android-tools}/bin/adb";
```
Comment (mweinelt) [inline pkgs/by-name/sp/spytrap-adb/package.nix:21]:
```suggestion
  env.SPYTRAP_ADB_BINARY = lib.getExe' android-tools "adb";
```

**Review by mweinelt — APPROVED** [review]
LGTM. Just a minor recommendation.

File: pkgs/by-name/sp/spytrap-adb/package.nix (hunk 6 above)
Comment (kpcyrd) [reply pkgs/by-name/sp/spytrap-adb/package.nix]: Thanks, done!

**Pull request merged**